marlowe-dashboard-client/src/Capability/Contract.purs

```diff
@@ -34,43 +34,43 @@ import WalletData.Types (Wallet)
 -- TODO (possibly): make `AppM` a `MonadError` and remove all the `runExceptT`s
 class
   Monad m <= ManageContract m where
-  activateContract :: ContractExe -> Wallet -> m (AjaxResponse PlutusAppId)
-  deactivateContract :: PlutusAppId -> m (AjaxResponse Unit)
-  getContractInstanceClientState :: PlutusAppId -> m (AjaxResponse (ContractInstanceClientState ContractExe))
-  getContractInstanceCurrentState :: PlutusAppId -> m (AjaxResponse (PartiallyDecodedResponse ActiveEndpoint))
-  getContractInstanceObservableState :: PlutusAppId -> m (AjaxResponse RawJson)
-  getContractInstanceHooks :: PlutusAppId -> m (AjaxResponse (Array (Request ActiveEndpoint)))
-  invokeEndpoint :: forall d. Encode d => PlutusAppId -> String -> d -> m (AjaxResponse Unit)
-  getWalletContractInstances :: Wallet -> m (AjaxResponse (Array (ContractInstanceClientState ContractExe)))
-  getAllContractInstances :: m (AjaxResponse (Array (ContractInstanceClientState ContractExe)))
-  getContractDefinitions :: m (AjaxResponse (Array (ContractSignatureResponse ContractExe)))
+  activateContract :: forall a. ContractActivationId a => a -> Wallet -> m (AjaxResponse PlutusAppId)
```

Comment:
looking at this usage I wonder if `Id` is the noun we are looking for... And looking at the endpoint below `deactivateContract`, I wonder if the `Activation` part is correct...

Maybe `ContractExe a` is the way to go, but I'd need to understand more all the options in order to name them.

Reply:
We can't use `ContractExe`, as that's already taken. Here, as elsewhere, I'm letting the Haskell names guide me. The `ContractActivationArgs` type is a record with a `caWallet` (contract activation wallet) and a `caID` (contract activation id) - and the latter is what my type here is for. I don't think it's a bad name: the type is intended to be a way of identifying the contract you want to activate. You either do that with the path to the executable (`ContractExe`) or some other built-in type.